src/core/ext/client_config/http_connect_handshaker.h

```diff
@@ -34,14 +34,20 @@
 #ifndef GRPC_CORE_EXT_CLIENT_CONFIG_HTTP_CONNECT_HANDSHAKER_H
 #define GRPC_CORE_EXT_CLIENT_CONFIG_HTTP_CONNECT_HANDSHAKER_H
 
+#include <stdbool.h>
+
 #include "src/core/lib/channel/handshaker.h"
 
-/// Does NOT take ownership of \a proxy_server or \a server_name.
-grpc_handshaker* grpc_http_connect_handshaker_create(const char* proxy_server,
-                                                     const char* server_name);
+/// Does NOT take ownership of \a server_name or \a authority. These two
+/// arguments will be the same most of the time except for load balanced
```

Comment:
I don't think that we've yet agreed exactly how we'll handle load-balanced connections via a proxy.  I'd like to conclude that discussion and come to agreement on a design before we start adding parameters to support one particular solution.

That having been said, you are certainly correct that the Host: header should be set to the server name in most cases.  So for now, let's just make that change, and we can deal with the rest once we've resolved the open issues about load-balancing via proxies.